planning/behavior_velocity_planner/occlusion-spot-design.md

```diff
@@ -4,104 +4,108 @@
 
 Note that this decision logic is still under development and needs to be improved.
 
-#### Occlusion Spot Public
+#### DetectionArea Polygon
 
-This module inserts safe velocity at the collision point estimated from the associated occlusion spot under assumption that the pedestrian possibly coming out of the occlusion spot.
-![brief](./docs/occlusion_spot/possible_collision_info.svg)
+This module considers TTV from pedestrian velocity and lateral distance to occlusion spot.
+TTC is calculated from ego velocity and acceleration and longitudinal distance until collision point using motion velocity smoother.
+To compute fast this module only consider occlusion spot whose TTV is less than TTC and only consider area within "max lateral distance".
 
-This module consider 3 policies that are very important for risk predicting system for occlusion spot.
+![brief](./docs/occlusion_spot/da.drawio.svg)
 
-1. "Passable" without deceleration
-   If ego vehicle speed is high enough to pass the occlusion spot and expected to have no collision with any objects coming out of the occlusion spot, then it's possible for ego vehicle to pass the spot without deceleration.
+#### Occlusion Spot Occupancy Grid Base
 
-2. "Predictable" with enough distance to occlusion
-   If ego vehicle has enough distance to the occlusion spot, then ego vehicle is going to slow down to the speed that is slow enough to stop before collision with full brake.
-   If ego vehicle pass the possible collision point, then ego vehicle is going to drive normally.
+This module considers any occlusion spot around ego path computed from the occupancy grid.
+Due to the computational cost occupancy grid is not high resolution and this will make occupancy grid noisy so this module add information of occupancy to occupancy grid map.
 
-3. "Unavoidable" without enough distance to occlusion spot
-   This module assumes the occlusion spot is detected stably far from the ego vehicle. Therefore this module can not guarantee the safety behavior for the occlusion spot detected suddenly in front of the ego vehicle. In this case, slow velocity that does not cause the strong deceleration is only applied.
+TODO: consider hight of obstacle point cloud to generate occupancy grid.
 
-#### Occlusion Spot Private
+##### Collision Free Judgement
 
-This module considers any occlusion spot around ego path computed from the occupancy grid.
+obstacle that can run out from occlusion should have free space until intersection from ego vehicle
```

Comment:
The words "dart out" ,  "jump out" and "run out" are used to describe a pedestrian entering the road from a occlusion spot. I think "jump out" is appropriate for entering from a occlusion spot, but what do you think?